Dockerfile

```diff
@@ -3,17 +3,13 @@ FROM golang:1.11.0-stretch
 
 RUN apt-get update && apt-get install -y make git unzip
 
-# should stay consistent with the version we use in Circle builds
-ARG PROTOC_VERSION=3.5.0
-# make a directory to do these operations in
-RUN export PROTOC_TMP_DIR=protoc && mkdir -p $PROTOC_TMP_DIR && cd $PROTOC_TMP_DIR \
-  # download the pre-built protoc binary
-  && curl --silent --show-error --location --output protoc.zip \
-    https://github.com/google/protobuf/releases/download/v$PROTOC_VERSION/protoc-$PROTOC_VERSION-linux-x86_64.zip \
-  # move the binary to /bin. move the well-known types ot /usr/local/include
-  && unzip protoc.zip && mv bin/protoc /bin/protoc && mv include/* /usr/local/include \
-  # remove all of the installation files
-  && cd .. && rm -rf $PROTOC_TMP_DIR
+# should stay consistent with the version in .circleci/config.yml
+ARG PROTOC_VERSION=3.6.1
+# download and install protoc binary and .proto files
+RUN curl --silent --show-error --location --output protoc.zip \
+  https://github.com/google/protobuf/releases/download/v$PROTOC_VERSION/protoc-$PROTOC_VERSION-linux-x86_64.zip \
+  && unzip -d /usr/local protoc.zip include/\* bin/\* \
+  && rm -f protoc.zip
```

Comment:
I would agree; maybe put it to `direct.mk`?

Reply:
The only problem is I'm waiting for https://github.com/docker/swarmkit/pull/2750 to be merged first as it modifies Makefiles and I'd like to avoid rebasing )

Reply:
Agreed on both accounts, reviewing 2570!